src/main/java/org/folio/orders/events/handlers/ReceiptStatusConsistency.java

```diff
@@ -1,162 +1,85 @@
 
 
   @Autowired
-  public ReceiptStatusConsistency(Vertx vertx, PurchaseOrderLineService purchaseOrderLineService) {
+  public ReceiptStatusConsistency(Vertx vertx, PieceStorageService pieceStorageService, PurchaseOrderLineService purchaseOrderLineService) {
     super(vertx.getOrCreateContext());
+    this.pieceStorageService = pieceStorageService;
     this.purchaseOrderLineService = purchaseOrderLineService;
   }
 
   @Override
   public void handle(Message<JsonObject> message) {
-    JsonObject messageFromEventBus = message.body();
-
+    var messageFromEventBus = message.body();
     logger.info("Received message body: {}", messageFromEventBus);
 
-    Map<String, String> okapiHeaders = org.folio.orders.utils.HelperUtils.getOkapiHeaders(message);
+    var okapiHeaders = getOkapiHeaders(message);
     var requestContext = new RequestContext(ctx, okapiHeaders);
-    List<Future<Void>> futures = new ArrayList<>();
-    Promise<Void> promise = Promise.promise();
-    futures.add(promise.future());
-
-    String poLineIdUpdate = messageFromEventBus.getString("poLineIdUpdate");
-    String query = String.format(PIECES_ENDPOINT, poLineIdUpdate, LIMIT);
-
-    // 1. Get all pieces for poLineId
-    getPieces(query, requestContext)
-      .compose(listOfPieces ->
-        // 2. Get PoLine for the poLineId which will be used to calculate PoLineReceiptStatus
-        purchaseOrderLineService.getOrderLineById(poLineIdUpdate, requestContext)
-          .map(poLine -> {
-            if (PoLineCommonUtil.isCancelledOrOngoingStatus(poLine)) {
-              promise.complete();
-              return null;
-            }
-            ReceiptStatus receivingStatus = calculatePoLineReceiptStatus(poLine, listOfPieces);
-            boolean statusUpdated = purchaseOrderLineService.updatePoLineReceiptStatusWithoutSave(poLine, receivingStatus);
-            if (statusUpdated) {
-              purchaseOrderLineService.saveOrderLine(poLine, requestContext)
-                .map(aVoid -> {
-                  // send event to update order status
-                  updateOrderStatus(poLine, okapiHeaders, requestContext);
-                  promise.complete();
-                  return null;
-                })
-                .onFailure(e -> {
-                  logger.error("The error updating poLine by id {}", poLineIdUpdate, e);
-                  promise.fail(e);
-                });
-            } else {
-              promise.complete();
-            }
-            return null;
-          })
-          .onFailure(e -> {
-            logger.error("The error getting poLine by id {}", poLineIdUpdate, e);
-            promise.fail(e);
-          }))
-      .onFailure(e -> {
-        logger.error("The error happened getting all pieces by poLine {}", poLineIdUpdate, e);
-        promise.fail(e);
-      });
 
-    // Now wait for all operations to be completed and send reply
-    completeAllFutures(futures, message);
-  }
```

Comment:
how simpler it become without promises, nice

Reply:
Yes, I suggest we refactor older code little by little in PRs as we will have much cleaner codebase